src/main/java/com/Acrobot/ChestShop/Listeners/Player/PlayerInteract.java

```diff
@@ -147,9 +156,15 @@ private static PreTransactionEvent preparePreTransactionEvent(Sign sign, Player
         boolean adminShop = ChestShopSign.isAdminShop(sign);
 
         // check if player exists in economy
-        if (!adminShop && !VaultListener.getProvider().hasAccount(account.getName())) {
-            player.sendMessage(Messages.prefix(Messages.NO_ECONOMY_ACCOUNT));
-            return null;
+        if (!adminShop) {
+            final boolean hasAccount = (com.Acrobot.ChestShop.ChestShop.isUsingReserve())?
+                ReserveListener.getProvider() != null &&
+                    ReserveListener.getProvider().hasAccount(account.getUuid()) :
+                VaultListener.getProvider().hasAccount(account.getName());
+            if(!hasAccount) {
+                player.sendMessage(Messages.prefix(Messages.NO_ECONOMY_ACCOUNT));
+                return null;
+            }
```

Comment:
Maybe this part could be a bit more streamlined by using an AccountCheckEvent which the economy listeners can listen?

This would get rid of the requirement to store which economy plugin is used and having to directly access different providers. Granted this should've been implemented from the beginning but wasn't really obvious until someone tried to add another economy plugin.